queries/flights.py

```diff
@@ -0,0 +1,502 @@
+
+QFlightsPrepLoc →
+    "til" QFlightsArrLoc_ef
+    | "frá" QFlightsDepLoc_þgf
+    | "í" QFlightsArrLoc_þgf
+    | "á" QFlightsArrLoc_þgf
+    | "á" QFlightsArrLoc_þf
+
+QFlightsArrLoc/fall →
+    Nl/fall
+
+QFlightsDepLoc/fall →
+    Nl/fall
+
+$tag(keep) QFlightsArrLoc/fall
+$tag(keep) QFlightsDepLoc/fall
+$score(+40) QFlights
+
+"""
+
+_LOCATION_ABBREV_MAP = {
+    "köben": "kaupmannahöfn",
+    "kef": "keflavík",
+    "álaborg": "aalborg",
+}
+
+_IATA_TO_AIRPORT_MAP = {
+    "aey": "akureyri",
+    "biu": "bíldudalur",
+    "egs": "egilsstaðir",
+    "gjr": "gjögur",
+    "gry": "grímsey",
+    "hfn": "hornafjörður",
+    "hzk": "húsavík",
+    "ifj": "ísafjörður",
+    "kef": "keflavík",
+    "rkv": "reykjavík",
+    "sak": "sauðárkrókur",
+    "tho": "þórshöfn",
+    "vey": "vestmannaeyjar",
+    "vpn": "vopnafjörður",
+}
+
+_AIRPORT_TO_IATA_MAP = {
+    "akureyri": "aey",
+    "bíldudalur": "biu",
+    "egilsstaðir": "egs",
+    "gjögur": "gjr",
+    "grímsey": "gry",
+    "hornafjörður": "hfn",
+    "húsavík": "hzk",
+    "ísafjörður": "ifj",
+    "keflavík": "kef",
+    "reykjavík": "rkv",
+    "sauðárkrókur": "sak",
+    "þórshöfn": "tho",
+    "vestmannaeyjar": "vey",
+    "vopnafjörður": "vpn",
+}
+
+# Day indices in accusative case
+_DAY_INDEX_ACC = {
+    1: "fyrsta",
+    2: "annan",
+    3: "þriðja",
+    4: "fjórða",
+    5: "fimmta",
+    6: "sjötta",
+    7: "sjöunda",
+    8: "áttunda",
+    9: "níunda",
+    10: "tíunda",
+    11: "ellefta",
+    12: "tólfta",
+    13: "þrettánda",
+    14: "fjórtánda",
+    15: "fimmtánda",
+    16: "sextánda",
+    17: "sautjánda",
+    18: "átjánda",
+    19: "nítjánda",
+    20: "tuttugasta",
+    21: "tuttugasta og fyrsta",
+    22: "tuttugasta og annan",
+    23: "tuttugasta og þriðja",
+    24: "tuttugasta og fjórða",
+    25: "tuttugasta og fimmta",
+    26: "tuttugasta og sjötta",
+    27: "tuttugasta og sjöunda",
+    28: "tuttugasta og áttunda",
+    29: "tuttugasta og níunda",
+    30: "þrítugasta",
+    31: "þrítugasta og fyrsta",
+}
+
+
+def QFlightsQuery(node, params, result):
+    # Set the query type
+    result.qtype = _FLIGHTS_QTYPE
+
+
+def QFlightsArrivalQuery(node, params, result):
+    result["departure"] = False
+
+
+def QFlightsDepartureQuery(node, params, result):
+    result["departure"] = True
+
+
+def QFlightsArrLoc(node, params, result):
+    result["to_loc"] = result._nominative
+
+
+def QFlightsDepLoc(node, params, result):
+    result["from_loc"] = result._nominative
+
+
+_ISAVIA_FLIGHTS_URL = (
+    "https://www.isavia.is/json/flight/?cargo=0&airport={0}"
+    "&dateFrom={1}&dateTo={2}&language=is&departures={3}"
+)
+
+_FLIGHTS_CACHE_TTL = 600  # seconds, ttl = 10 mins
+
+# Cache for flights either departing or arriving
+_FLIGHT_CACHE = cachetools.TTLCache(maxsize=2, ttl=_FLIGHTS_CACHE_TTL)
+
+
+def _fetch_flight_data(
+    from_date: datetime, to_date: datetime, iata_code: str, departing: bool
+) -> List[Dict[str, Any]]:
```

Comment:
Consider defining a type early in the file: `FlightList = List[Dict[str, Any]]`

Reply:
Fixed!